src/core/lib/surface/channel_init.cc

```diff
@@ -426,39 +569,9 @@ bool ChannelInit::Filter::CheckPredicates(const ChannelArgs& args) const {
 
 bool ChannelInit::CreateStack(ChannelStackBuilder* builder) const {
   const auto& stack_config = stack_configs_[builder->channel_stack_type()];
-  for (const auto& filter : stack_config.filters) {
-    if (SkipV2(filter.version)) continue;
-    if (!filter.CheckPredicates(builder->channel_args())) continue;
-    builder->AppendFilter(filter.filter);
-  }
-  int found_terminators = 0;
-  for (const auto& terminator : stack_config.terminators) {
-    if (!terminator.CheckPredicates(builder->channel_args())) continue;
-    builder->AppendFilter(terminator.filter);
-    ++found_terminators;
-  }
-  if (found_terminators != 1) {
-    std::string error = absl::StrCat(
-        found_terminators,
-        " terminating filters found creating a channel of type ",
-        grpc_channel_stack_type_string(builder->channel_stack_type()),
-        " with arguments ", builder->channel_args().ToString(),
-        " (we insist upon one and only one terminating "
-        "filter)\n");
-    if (stack_config.terminators.empty()) {
-      absl::StrAppend(&error, "  No terminal filters were registered");
-    } else {
-      for (const auto& terminator : stack_config.terminators) {
-        absl::StrAppend(&error, "  ", terminator.name, " registered @ ",
-                        terminator.registration_source.file(), ":",
-                        terminator.registration_source.line(), ": enabled = ",
-                        terminator.CheckPredicates(builder->channel_args())
-                            ? "true"
-                            : "false",
-                        "\n");
-      }
-    }
-    LOG(ERROR) << error;
+  if (!SelectThenMergeFilters(builder, stack_config.filters,
+                              stack_config.fused_filters, false) ||
+      !SelectThenMergeFilters(builder, stack_config.terminators, {}, true)) {
```

Comment:
I think we can remove SelectThenMergeFilters -- we don't need fusion on terminators since they're never v3 things.